static/js/tools/map/fetcher/breadcrumb_denom_stat.ts

```diff
@@ -20,59 +20,86 @@
 
 import axios from "axios";
 import _ from "lodash";
-import { Dispatch } from "react";
+import { Dispatch, useContext, useEffect } from "react";
 
 import {
   EntitySeriesWrapper,
   SeriesApiResponse,
 } from "../../../shared/stat_types";
 import { stringifyFn } from "../../../utils/axios";
-import { ChartDataType, ChartStoreAction } from "../chart_store";
+import { ChartDataType, ChartStore, ChartStoreAction } from "../chart_store";
+import { useCalculateRatio } from "../compute_hook";
+import { Context } from "../context";
+import { useBreadcrumbDenomStatReady } from "../ready_hook";
 
-export function fetchBreadcrumbDenomStat(
-  entities: string[],
-  statVar: string,
+export function useFetchBreadcrumbDenomStat(
+  chartStore: ChartStore,
   dispatch: Dispatch<ChartStoreAction>
 ): void {
-  const action: ChartStoreAction = {
-    type: ChartDataType.BREADCRUMB_DENOM_STAT,
-    context: {
-      placeInfo: {
-        enclosingPlace: {
-          // entities are the parent places + the enclosing place.
-          // so the last element is the enclosing place.
-          dcid: entities[entities.length - 1],
-          name: "",
+  const { placeInfo, statVar } = useContext(Context);
+  const breadcrumbDenomStatReady = useBreadcrumbDenomStatReady(chartStore);
```

Comment:
also curious why we're checking if breadcrumbDenomStatReady? will there be cases where we enter this hook but we've already fetched the relevant data?

Reply:
This is the case where user click on "per capita" then choose another statvar. In this case, we don't need to fetch denom stat again since it's already there.

Basically these hooks holds the logic with **Ready() to determine if data is necessary to fetch.